model/Dependency.js

```diff
@@ -1,6 +1,76 @@
+// Matches semantic versions like:
+// <major>
+// <major>.<minor>
+// <major>.<minor>.<patch>
+// <major>.<minor>.<patch>-<tag>
+// <major>.<minor>.<patch>@<tag>
+//
+// Examples:
+//
+// Version string: "~> 4.3"
+// Group 1: 4
+// Group 2: 3
+// Group 3: <undefined>
+// Group 4: <undefined>
+//
+// Version string: v4@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b
+// Group 1: 4
+// Group 2: <undefined>
+// Group 3: <undefined>
+// Group 4: 44c2b7a8a4ea60a981eaca3cf939b5f4305c123b
+//
+// Version string: ^7.20.2
+// Group 1: 7
+// Group 2: 20
+// Group 3: 2
+// Group 4: <undefined>
+//
+// Version string: v7.0.0-bridge.0
+// Group 1: 7
+// Group 2: 0
+// Group 3: 0
+// Group 4: bridge.0
+//
+// Versions for which no semantic versions can be derived should be treated as an opaque tag.
+const MAJOR_MINOR_PATCH_TAG_REGEX = /(\d+)(?:\.(\d+)(?:\.(\d+))?)?(?:[-@](\S+))?/;
+
 export class Dependency {
   constructor(name, version) {
     this.name = name;
-    this.version = version;
+
```

Comment:
I think it would be good to unit test this constructor now it has non-trivial functionality

Reply:
I've just added some tests for this.